src/analyze/analyze-time-data.c

```diff
@@ -254,14 +287,15 @@ int acquire_time_data(sd_bus *bus, bool require_finished, UnitTimes **out) {
                 return bus_log_parse_error(r);
 
         while ((r = bus_parse_unit_info(reply, &u)) > 0) {
-                UnitTimes *t;
+                _cleanup_(unit_times_donep) UnitTimes *t = NULL;
```

Comment:
Why we need to clear `t` here on failure??

Reply:
Otherwise we will leak memory since it contains references. In a previous version of this patch, I forgot some of those, and LeakSanitizer did not like me.